envi/common.py

```diff
@@ -1,18 +1,25 @@
 import logging
 
 EMULOG = 9
-LOG_LEVELS = ('EMULOG', 'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL')
+LOG_LEVELS = (
+        'CRITICAL',
+        'ERROR',
+        'WARNING',
+        'INFO',
+        'DEBUG',
+        'EMULOG',
+        )
 LOG_FORMAT = '%(asctime)s:%(levelname)s:%(name)s: %(message)s' \
              '[%(filename)s:%(funcName)s:%(lineno)s:%(threadName)s]'
 
+logging.basicConfig(level=logging.WARNING, format=LOG_FORMAT)
+logging.addLevelName(EMULOG, 'EMULOG')
 
 def setLogging(logger, level=None):
-    if level:
+    if level is not None:
         level = str(level).upper()
         if level not in LOG_LEVELS:
             raise ValueError('Invalid log level of %r' % level)
     else:
         level = 'ERROR'
-    logging.basicConfig(level=level, format=LOG_FORMAT)
```

Comment:
Yea, I'll echo what Willi said, why move this line out of the setLogging call?

Reply:
those are two different things.  
i'll change it so that vivbin and vdbbin actually call basicConfig (or move this to another function).
as for "out of the setLogging call," i want to allow the changing of logging level without re-calling basicConfig().

Reply:
So I don't think those are two different things. Moving the basicConfig call out of this function to the global does accidentally override the logging levels set by any downstream consumer of viv, and since people do use viv as a library, that does mess up their use cases.

That and I'm not quite sure about what use case you have for your second statement of ` i want to allow the changing of logging level without re-calling basicConfig().` While calling setLogging does call basicConfig and is inefficient after the first time, it gives us a nice level of uniformity for our log messages that I tend to like. Beyond that, if we're dynamically setting the logging enough for that to impact performance, I have concerns. Well, that, and since this all just logging, calling setLevel is fairly trivial and it's not like we've precluded the use of that. So I'm not entirely sure what your concern here is?

Reply:
resolved out-of-band